Source/bmalloc/libpas/src/libpas/pas_deallocate.c

```diff
@@ -64,6 +65,33 @@ bool pas_try_deallocate_known_large(void* ptr,
     return true;
 }
 
+bool pas_try_deallocate_pgm_large(void* ptr,
+                                  const pas_heap_config* config_ptr)
+{
+    uintptr_t begin = (uintptr_t)ptr;
+    pas_heap_lock_lock();
+    if (config_ptr->pgm_enabled && pas_probabilistic_guard_malloc_check_exists(begin)) {
+        pas_probabilistic_guard_malloc_deallocate((void *) begin);
+        pas_heap_lock_unlock();
+        return true;
+    }
+    pas_heap_lock_unlock();
+    return false;
```

Comment:
If `config_ptr->pgm_enabled` is only written once (on process launch, presumably), does it make sense to add an early return check since we wouldn't have to take the `pas_heap_lock` to check it?

(My OCD wants to combine the function calls to `pas_heap_lock_unlock()`, so I changed that as well.  Not sure if it's more CPU efficient, but should at least be more space-efficient.)

```suggestion
    if (!config_ptr->pgm_enabled)
        return false;
    pas_heap_lock_lock();
    bool result = pas_probabilistic_guard_malloc_check_exists(begin);
    if (result)
        pas_probabilistic_guard_malloc_deallocate((void *) begin);
    pas_heap_lock_unlock();
    return result;
```

Reply:
Fixed. Thanks